Samples/ios/SampleApp/SampleAppiOS/SampleAppDelegate.m

```diff
@@ -0,0 +1,135 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import <MSAL/MSAL.h>
+
+#import "SampleAppDelegate.h"
+#import "SampleMSALUtil.h"
+#import "SampleMainViewController.h"
+#import "SampleLoginViewController.h"
+
+@interface SampleAppDelegate ()
+{
+    UIViewController *_rootController;
+    UIViewController *_currentController;
+}
+
+@end
+
+@implementation SampleAppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+    // The MSAL Logger should be set as early as possible in the app launch sequence, before any MSAL
+    // requests are made.
+    [SampleMSALUtil setup];
+    
+    UIWindow* window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
+    self.window = window;
+    
+    _rootController = [UIViewController new];
+    
+    if ([SampleMSALUtil currentUser:nil])
+    {
+        [self setCurrentViewController:[SampleMainViewController sharedViewController]];
+    }
+    else
+    {
+        [self setCurrentViewController:[SampleLoginViewController sharedViewController]];
+    }
+    
+    [window setRootViewController:_rootController];
+    [window setBackgroundColor:[UIColor whiteColor]];
+    [window makeKeyAndVisible];
+    
+    return YES;
+}
+
+- (BOOL)application:(UIApplication *)app
+            openURL:(NSURL *)url
+            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options
+{
+    if ([MSALPublicClientApplication isMSALResponse:url])
+    {
+        [MSALPublicClientApplication handleMSALResponse:url];
+        return YES;
```

Comment:
I want to emphasize taht you should return immediately after calling handleMSALResponse in case someone were to copy & paste this into their own code.